src/org/thoughtcrime/securesms/loki/SeedActivity.kt

```diff
@@ -44,11 +53,18 @@ class SeedActivity : BaseActionBarActivity() {
         setContentView(R.layout.activity_seed)
         setUpLanguageFileDirectory()
         updateSeed()
+        updateUI()
         copyButton.setOnClickListener { copy() }
-        toggleModeButton.setOnClickListener { toggleMode() }
+        toggleRestoreModeButton.setOnClickListener { mode = Mode.Restore }
+        toggleRegisterModeButton.setOnClickListener { mode = Mode.Register }
+        toggleLinkModeButton.setOnClickListener { mode = Mode.Link }
         registerOrRestoreButton.setOnClickListener { registerOrRestore() }
         Analytics.shared.track("Seed Screen Viewed")
     }
+
+    override fun onDestroy() {
+        super.onDestroy()
```

Comment:
?

Reply:
woops, this was when i was still thinking about where to put the logic, we can remove this.